rs/ic_os/config/src/main.rs

```diff
@@ -205,6 +294,203 @@ pub fn main() -> Result<()> {
+            };
+
+            let guestos_config_json_path = Path::new(&guestos_config_json_path);
+            serialize_and_write_config(guestos_config_json_path, &guestos_config)?;
+
+            println!(
+                "GuestOSConfig has been written to {}",
+                guestos_config_json_path.display()
+            );
+
+            Ok(())
+        }
+        Some(Commands::GenerateTestnetConfig(clap_args)) => {
+            // Convert `clap_args` into `GenerateTestnetConfigArgs`
+            let args = GenerateTestnetConfigArgs {
+                ipv6_config_type: clap_args.ipv6_config_type,
+                deterministic_prefix: clap_args.deterministic_prefix,
+                deterministic_prefix_length: clap_args.deterministic_prefix_length,
+                deterministic_gateway: clap_args.deterministic_gateway,
+                fixed_address: clap_args.fixed_address,
+                fixed_gateway: clap_args.fixed_gateway,
+                ipv4_address: clap_args.ipv4_address,
+                ipv4_gateway: clap_args.ipv4_gateway,
+                ipv4_prefix_length: clap_args.ipv4_prefix_length,
+                domain_name: clap_args.domain_name,
+                mgmt_mac: clap_args.mgmt_mac,
+                deployment_environment: clap_args.deployment_environment,
+                elasticsearch_hosts: clap_args.elasticsearch_hosts,
+                elasticsearch_tags: clap_args.elasticsearch_tags,
+                nns_public_key_exists: clap_args.nns_public_key_exists,
+                nns_urls: clap_args.nns_urls,
+                node_operator_private_key_exists: clap_args.node_operator_private_key_exists,
+                use_ssh_authorized_keys: clap_args.use_ssh_authorized_keys,
+                inject_ic_crypto: clap_args.inject_ic_crypto,
+                inject_ic_state: clap_args.inject_ic_state,
+                inject_ic_registry_local_store: clap_args.inject_ic_registry_local_store,
+                backup_retention_time_seconds: clap_args.backup_retention_time_seconds,
+                backup_purging_interval_seconds: clap_args.backup_purging_interval_seconds,
+                malicious_behavior: clap_args.malicious_behavior,
+                query_stats_epoch_length: clap_args.query_stats_epoch_length,
+                bitcoind_addr: clap_args.bitcoind_addr,
+                jaeger_addr: clap_args.jaeger_addr,
+                socks_proxy: clap_args.socks_proxy,
+                hostname: clap_args.hostname,
+                generate_ic_boundary_tls_cert: clap_args.generate_ic_boundary_tls_cert,
+            };
+
+            generate_testnet_config(args, clap_args.guestos_config_json_path)
+        }
+        // TODO(NODE-1519): delete UpdateGuestosConfig and UpdateHostosConfig after moved to new config format
+        Some(Commands::UpdateGuestosConfig { output_file }) => update_guestos_config(output_file),
+        Some(Commands::UpdateHostosConfig {
```

Comment:
Note that UpdateHostosConfig only updates config. It does not "downgrade" config, unlike UpdateGuestosConfig which does both. 

The reason for this is that we can just continue to pass the old config to HostOS (deployment.template and config.ini files).

Reply:
The deployment.template and config.ini files? Yes! Because we will continue to pass the old config to HostOS even as we use the new config (which we don't do for GuestOS), and the config partition persists between reboots

Reply:
Perfect! What makes the process different for GuestOS? SetupOS already has the old configs, so it is trivial to keep both formats around?

Reply:
Update: I've update the config revamp integration to continue passing the old GuestOS config, so we no longer need the GuestOS "downgrade-config" logic. I've commented it out.